kedro-datasets/kedro_datasets/snowflake/snowpark_dataset.py

```diff
@@ -199,45 +208,76 @@ def _get_session(connection_parameters) -> sp.Session:
         """
         try:
             logger.debug("Trying to reuse active snowpark session...")
-            session = sp.context.get_active_session()
-        except sp.exceptions.SnowparkSessionException:
+            session = sp_context.get_active_session()
+        except sp_exceptions.SnowparkSessionException:
             logger.debug("No active snowpark session found. Creating...")
-            session = sp.Session.builder.configs(connection_parameters).create()
+            session = Session.builder.configs(connection_parameters).create()
         return session
 
     @property
-    def _session(self) -> sp.Session:
-        return self._get_session(self._connection_parameters)
+    def _session(self) -> Session:
+        """
+        Retrieve or create a session.
+
+        Returns:
+            Session: The current session associated with the object.
+        """
+        if not self.__session:
+            self.__session = self._get_session(self._connection_parameters)
+        return self.__session
 
-    def load(self) -> sp.DataFrame:
-        table_name: list = [
-            self._database,
-            self._schema,
-            self._table_name,
-        ]
+    def load(self) -> DataFrame:
+        """
+        Load data from a specified database table.
 
-        sp_df = self._session.table(".".join(table_name))
-        return sp_df
+        Returns:
+            DataFrame: The loaded data as a Snowpark DataFrame.
+        """
+        return self._session.table(self._validate_and_get_table_name())
 
-    def save(self, data: sp.DataFrame) -> None:
-        table_name = [
-            self._database,
-            self._schema,
-            self._table_name,
-        ]
+    def save(self, data: pd.DataFrame | DataFrame) -> None:
+        """
+        Save data to a specified database table.
 
-        data.write.save_as_table(table_name, **self._save_args)
+        Args:
+            data (pd.DataFrame | DataFrame): The data to save.
+        """
+        if isinstance(data, pd.DataFrame):
+            data = self._session.create_dataframe(data)
```

Comment:
Maybe we should add an additional check and raise an error here, like this:

```python
# Check if the input is a Pandas DataFrame and convert it to Snowpark DataFrame
if isinstance(data, pd.DataFrame):
    # Convert the Pandas DataFrame to a Snowpark DataFrame
    snowpark_df = self._session.create_dataframe(data)
elif isinstance(data, DataFrame):
    # If it's already a Snowpark DataFrame, use it as is
    snowpark_df = data
else:
    raise DatasetError(f"Data of type {type(data)} is not supported for saving.")
```

This ensures we handle different types of DataFrames appropriately, as we're currently allowing not only Snowpark DataFrames but other DataFrame types as well.